### PR TITLE
Adding working GitHub action workflow

### PR DESCRIPTION
**Pull request by getvictor**
Added GitHub action workflow.
@lucasmrod Please review README.md, workflow.yml, and workflow.sh files

Also, updated indentation to be consistent.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,23 +2,30 @@
...
+3. Create your own GitHub repository.
 
-4. Add Fleet's open-source [GitHub action](https://github.com/fleetdm/fleet-gitops) to your repo. Now, when anyone makes a change to any file, the action will run and update Fleet.
+4. Add `FLEET_URL` and `FLEET_API_KEY` to your GitHub repository secrets.
+
+5. Update the `env` section of your [.github/workflows/workflow.yml](https://github.com/fleetdm/fleet-gitops/blob/main/.github/workflows/workflow.yml) file for all the environment variables that are mentioned in your `default.yml` and `teams/*.yml` YAML files. For the environment variables that map to secrets, add the required secrets to your GitHub repository secrets.
+
+6. Push your cloned and modified code to your repo.
+
+7. In GitHub, enable the `Apply latest configuration to Fleet` GitHub Actions workflow, and run workflow manually. Now, when anyone pushes a new commit, the action will run and update Fleet. In addition, the workflow will run every 24 hours to ensure that the Fleet configuration is in sync with your git repo.
```
Comment (noahtalerman) [inline README.md:23]:
```suggestion
7. In GitHub, enable the `Apply latest configuration to Fleet` GitHub Actions workflow, and run workflow manually. Now, when anyone pushes a new commit, the action will run and update Fleet.
```

I see what you're after with this! If any change is made outside of the repo (UI/API/fleetctl), Fleet will get reset (sync w/ the repo).

I think we can cut this feature for v1. 

Why? I think it adds more UX complexity. More for the user to keep track of and more for us to explain.

The user can manually trigger the GitHub action if they want to sync right? If so, I think starting w/o the automated run every 24 hours is simpler. The action runs on merge and when you trigger it.

We can get fancy later.

**Review by noahtalerman — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (getvictor) [reply README.md]: Done

**Review by noahtalerman — APPROVED** [review]
YAML interface and README look good to me!

I'll lean on y'all to review the workflow and file and bash script.

**Comment by noahtalerman** [comment]
Hey @getvictor I think we should make you the CODEOWNER of the workflow file and bash script so you get pinged anytime someone wants to make a change. 

What do you think?

I can be CODEOWNERS of the README + YAML files + lib/

If you're cool with that can you please open a PR to CODEOWNERS?

**Review by lucasmrod — APPROVED** [review]
Looks great!

File: workflow.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+FLEETCTL="${FLEETCTL:-fleetctl}"
+
+# Validate that global file contains org_settings
+grep -Exq "^org_settings:.*" $GLOBAL_FILE
+
+if compgen -G ./teams/*.yml > /dev/null; then
+  # Validate that every team has a unique name.
+  # This is a limited check that assumes all team files contain the phrase: `name: <team_name>`
+  ! perl -nle 'print $1 if /^name:\s*(.+)$/' ./teams/*.yml | sort | uniq -d | grep . -cq
+fi
```
Comment (lucasmrod) [inline workflow.sh:19]: Q: Could you explain what `compgen -G ./teams/*.yml` does? (It returns almost all files in the current directory for me on macOS)

Comment (lucasmrod) [reply workflow.sh:19]: Nevermind, I was executing it incorrectly... (TIL!)

**Pull request merged**